btcec/privkey.go

```diff
@@ -59,7 +60,28 @@ func (p *PrivateKey) ToECDSA() *ecdsa.PrivateKey {
 // is deterministic (same message and same key yield the same signature) and canonical
 // in accordance with RFC6979 and BIP0062.
 func (p *PrivateKey) Sign(hash []byte) (*Signature, error) {
-	return signRFC6979(p, hash)
+	return signRFC6979(p, hash, 0)
+}
+
+// SignCanonical goes through signatures and returns only a canonical
+// representations.  This matches the EOS blockchain expectations.
+func (p *PrivateKey) SignCanonical(curve *KoblitzCurve, hash []byte) ([]byte, error) {
```

Comment:
`Canonical` seems quite general here, while it's just EOS-general. Should be renamed to reflect that.

Also, some references to the algo would be nice. F.e. the magic number 25 should be explained somehow.